TileDB.Cloud/Array.cs

```diff
@@ -0,0 +1,288 @@
+                var policyResult = Client.GetInstance().GetRetryAsyncPolicyWrap().ExecuteAndCaptureAsync(
+                    () => Udf.UdfApi.SubmitUDFAsync(arrayNamespace, arrayName, udf,
+                        chargedOrg ?? GetDefaultChargedNamespace()));
+                policyResult.ContinueWith(t => CheckPolicyResult(t.Result));
+                return policyResult;
+            }
+
+            /// <summary>
+            /// Execute a multi-array UDF
+            /// </summary>
+            /// <param name="udfUri">TileDB uri for UDF to execute</param>
+            /// <param name="arrayList">Constructed ArrayList object</param>
+            /// <param name="arrayNamespace">Namespace arrays are in</param>
+            /// <param name="args">JSON formatted arguments to pass to UDF</param>
+            /// <returns>Stream containing multi-array UDF result</returns>
+            public static Stream ApplyMultiArray(
+                string udfUri, ArrayList arrayList, string arrayNamespace, string? args = null)
+            {
+                var udf = new ArrayUdf(udfUri, arrayList, args!);
+                var policyResult = Client.GetInstance().GetRetryPolicyWrap().ExecuteAndCapture(
+                    () => Udf.UdfApi.SubmitMultiArrayUDF(arrayNamespace, udf));
+                CheckPolicyResult(policyResult);
+                return policyResult.Result;
+            }
+
+            /// <summary>
+            /// Execute a multi-array UDF asynchronously
+            /// </summary>
+            /// <param name="udfUri">TileDB uri for UDF to execute</param>
+            /// <param name="arrayList">Constructed ArrayList object</param>
+            /// <param name="arrayNamespace">Namespace arrays are in</param>
+            /// <param name="args">JSON formatted arguments to pass to UDF</param>
+            /// <returns>Asynchronous Task containing multi-array UDF result Stream</returns>
+            public static Task<PolicyResult<Stream>> ApplyMultiArrayAsync(
+                string udfUri, ArrayList arrayList, string arrayNamespace, string? args = null)
+            {
+                var udf = new ArrayUdf(udfUri, arrayList, args!);
+                var policyResult = Client.GetInstance().GetRetryAsyncPolicyWrap().ExecuteAndCaptureAsync<Stream>(
+                    () => Udf.UdfApi.SubmitMultiArrayUDFAsync(arrayNamespace, udf));
+                policyResult.ContinueWith(t => CheckPolicyResult(t.Result));
+                return policyResult;
+            }
+        }
+
+        public class ArrayList
+        {
+            public List<UDFArrayDetails> Arrays { get; set; }
+
+            public ArrayList()
+            {
+                Arrays = new List<UDFArrayDetails>();
+            }
+
+            public ArrayList(List<UDFArrayDetails> arrayList)
+            {
+                Arrays = arrayList;
+            }
+
+            public void Add(string arrayUri, List<dynamic> ranges, List<string> buffers, Layout layout)
```

Comment:
Using `dynamic` greatly troubles me. It destroys type safety, it is slow and uses reflection. We might need to make it generic or something; needs more thought. 🤔

Reply:
Upon second thought we might be able to replace `dynamic` with `object`. These are the same type under the hood; the difference is that the former enables dynamic typing. The example code should definitely switch unless it needs it (doubt), but changing it in the sources might need more thought.